flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableFactory.java

```diff
@@ -49,7 +47,6 @@ public List<String> supportedProperties() {
     @Override
     public TableSource createTableSource(TableSourceFactory.Context context) {
         CatalogTable table = checkNotNull(context.getTable());
-        Preconditions.checkArgument(table instanceof CatalogTableImpl);
```

Comment:
why we need remove this check?

Reply:
Because the new created table is a instance of ResolvedCatalogTable which could provide `Schema`.